src/pages/login/Login.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useCallback } from 'react';
+import { Button, Form, Input } from 'antd';
+import { useRootDispatch, useRootSelector } from '../../store';
+import { loginUser } from '../../store/user';
+import { requiredPasswordRule, requiredUsernameRule } from './form-rules';
+
+type AntdStore = { [key: string]: any }; // Antd doesn't seem to export this.
+type FormSchema = { username: string; password: string };
+
+export const LoginRoute = (): JSX.Element => {
+  const dispatch = useRootDispatch();
+
+  const onFinish = useCallback(
+    (store: AntdStore) => {
+      const values = store as FormSchema;
+      dispatch(loginUser(values as FormSchema));
+    },
+    [dispatch]
+  );
+
+  return (
+    <div className="container card">
+      <Form layout="vertical" name="basic" onFinish={onFinish}>
+        <Form.Item
+          label="Username"
+          name="username"
+          rules={[requiredUsernameRule]}
+        >
+          <Input size="large" />
+        </Form.Item>
```

Comment:
I noticed this just now reviewing my own code: I'd probably want to find out if we'd prefer renaming this field to `Email`. I think calling it `Username` and then expecting an email address likely confuses users.